### PR TITLE
Add github CI verification

### LINKED ISSUE #50: Change CI from Travis to Github Actions
The issue should speak for itself.

### PR DESCRIPTION
**Pull request by Laguna1989**
Will fix #50 

Please see the green verification from the master branch on my fork: 
https://github.com/Laguna1989/tileson/actions/runs/1972660812

Please also note the comments in the files in this PR.

### CODE REVIEW

File: tests/CMakeLists.txt [hunk 1]
```diff
@@ -23,5 +23,6 @@ elseif(APPLE)
     endif()
 elseif(WINDOWS)
     message("WINDOWS!")
+    target_compile_options(tileson_tests PUBLIC /bigobj)
```
Comment (Laguna1989) [inline tests/CMakeLists.txt:26]: The windows build on github actions was complaining about the /bigobj flag missing, that is why I added it. It does not seem to be required for local builds. 
As this only changes the compile options of the test, it does not seem critical.

File: .github/workflows/test_verification.yml [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+
+      - name: Test
+        working-directory: ${{github.workspace}}/build/tests/
+        run: ./tileson_tests --order rand
+
+  Linux_clang:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        clangver: [ 8, 11 ]
```
Comment (Laguna1989) [inline .github/workflows/test_verification.yml:50]: note: clang++ version 12 and 13 are explicitly not added. With this version of clang it seems that there are std::bad_alloc exceptions raised from multiple test cases, including the benchmark test and the "Parse a whole COMPRESSED map by file" test case in `tests_main.cpp`.

* A local clang++-10 version works fine. (unfortunately I cannot upgrade to a newer clang locally)
* The issue seems to be consistently appearing over multiple runs on github actions.
* The issue is also on osx, which uses clang-13 (see comment below).
* Splitting up the benchmark test case to reduce memory consumption did not solve the issue. 
* The bad alloc seems to originate somewhere in the memory LZMA code, as I could not reproduce it locally, debugging was not possible.

File: .github/workflows/test_verification.yml [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+
+      - name: Build
+        working-directory: ${{github.workspace}}/build
+        run: cmake --build .
+
+      - name: Test
+        working-directory: ${{github.workspace}}/build
+        run: tests/Debug/tileson_tests.exe --order rand
+
+#  Mac_clang:
```
Comment (Laguna1989) [inline .github/workflows/test_verification.yml:116]: osx is currently commented out, as of clang++-12/clang++-13 issues. I do not know how to install specific clang versions on osx.

File: .github/workflows/test_verification.yml (hunk 2 above)
Comment (SSBMTonberry) [reply .github/workflows/test_verification.yml:50]: Thanks!
The `std::bad_alloc` exceptions sounds strange, but if they are not produced by `Tileson` itself I wouldn't worry too much about it. If you found a specific cause to this, I would be interested in adding a fix in the `PocketLzma` library, though.

File: .github/workflows/test_verification.yml (hunk 3 above)
Comment (SSBMTonberry) [reply .github/workflows/test_verification.yml:116]: I created a project some while ago to test Github Actions, where I also struggled to get OSX to fork with Clang, but `GCC` with OSX worked fine. I guess Apple is just Apple, and have done some weird changes with Clang and how it locates standard libraries. For reference, that project can be found here:
https://github.com/SSBMTonberry/github-actions-tests

Comment (SSBMTonberry) [reply .github/workflows/test_verification.yml:116]: Oh, and it's fine that OSX is not included for now. Anything is better than nothing after all 🙂

File: tests/CMakeLists.txt (hunk 1 above)
Comment (SSBMTonberry) [reply tests/CMakeLists.txt:26]: Sounds like a weird and unnecessary requirement by Windows, but if it solves an issue on Windows by adding this option, I'll allow it 😛

**Comment by SSBMTonberry** [comment]
Thank you very much for implementing this feature! I've not got too much time working on `Tileson` lately, and `Travis` is not relly doing its job anymore, so much appreciated! 😄

**Review by SSBMTonberry — APPROVED** [review]

**Comment by Laguna1989** [comment]
Thank you for merging the PR. Always glad if I can help.

I did take a look at the LZMA issue and I can now reproduce it locally with clang++-12 on WSL.

Some further observations:
* This only happens on release builds
* Debug builds are fine and do not show the issue
* With printf-debugging I could nail it down to [this call](https://github.com/SSBMTonberry/tileson/blob/5dc17aa524375cbe1ff5a2f5520bee6ca83cda98/include/common/Lzma.hpp#L43)
* When I add printf-debug statements in `StatusCode PocketLzma::decompress(const uint8_t *input, const size_t inputSize, std::vector<uint8_t> &output)`, the error vanishes. Seems like a Heissenbug. :(

**Pull request merged**